server/streamable_http.go

```diff
@@ -294,10 +300,19 @@ func (s *StreamableHTTPServer) handlePost(w http.ResponseWriter, r *http.Request
 	// Write response
 	mu.Lock()
 	defer mu.Unlock()
+	// close the done chan before unlock
+	defer close(done)
 	if ctx.Err() != nil {
 		return
 	}
-	if upgraded {
+	if session.upgradeToSSE.Load() {
```

Comment:
Nit: Maybe add some inline comments to the entire `streamable_http.go` file for the changes you made?

Reply:
Thanks for your review. comments are added as suggested.